Library/Client.ts

```diff
@@ -236,6 +259,32 @@ class Client {
         let array = seq.split(":");
         return [array[0], parseInt(array[1])];
     }
+
+    private runTelemetryProcessors(envelope: ContractsModule.Contracts.Envelope): boolean {
+        var accepted = true;
+        var telemetryProcessorsCount = this._telemetryProcessors.length;
+
+        if (telemetryProcessorsCount === 0) {
+            return accepted;
+        }
+
+        try {
+            for (var i = 0; i < telemetryProcessorsCount; ++i) {
+                var processor = this._telemetryProcessors[i];
+                if (processor) {
+                    if (processor.apply(null, [envelope]) === false) {
+                        accepted = false;
+                        break;
+                    }
+                }
+            }
+        } catch (error) {
+            accepted = false;
+            Logging.warn("One of telemetry processors failed, telemetry item will not be sent.", error, envelope);
```

Comment:
Good point, IMHO if a first processor throws, there is no need to run other processors because the data may already be corrupted. I'll move the try/catch inside the _for_ loop, and leave the _break_ statement.